span.go

```diff
@@ -374,6 +387,13 @@ func (s *Span) reportSelfTime() {
 
 	s.tx.TransactionData.mu.Lock()
 	defer s.tx.TransactionData.mu.Unlock()
+	s.reportSelfTimeLockless(endTime)
+}
+
+// reportSelfTimeLockless is used by all spans (composite and regular spans) to
+// report their timers. Since the composite end is triggered by a span of the
+// same kind or its parent ending, the upstream locks are already acquired.
```

Comment:
reportSelfTime also locks TransactionData to update its childrenTimer. Are we doing that in span.End()?

Reply:
great catch, I totally missed that one, I'll review the locks again